content/docs/building-apps/add-stellar-assets.mdx

```diff
@@ -0,0 +1,185 @@
+---
+title: Set Up a Custodial Account
+order: 45
+---
+
+import { CodeExample } from "components/CodeExample";
+import { Alert } from "components/Alert";
+
+This guide describes how to add tokens from the Stellar network to your custodial service. First, we walk through adding Stellar native asset, lumens. Following that, we describe how to add other tokens. This example uses Node.js and the [JS Stellar SDK](https://github.com/stellar/js-stellar-sdk), but it should be easy to adapt to other languages.
+
+
+## Account Setup
+
+### Pooled account
+Most custodial services, including crypto exchanges, choose to use a single pooled Stellar account to handle transactions on behalf of their users instead of creating a new Stellar account for each customer. Generally, they keep track of their customers in an internal database and use the memo field of a Stellar transaction to map an incoming payment to the corresponding internal customer ID.
+
+The benefits of using a pooled account are low cost, no base reserves needed for each account; and low complexity, you only need to manage one account key. However, with a pooled account, it is your responsibility to manage all individual balances and payments. You can no longer rely on Stellar ledger to accumulate value, handle errors and atomicity, and manage transactions on a customer-by-customer basis.
+
+
+## Code Framework
+Use this code framework to integrate Stellar into your custodial service. For this guide, we use placeholder functions for reading/writing to your internal customer database. Each organization will architect their infrastructure differently, so we abstract away those details. Here are the functions we'll assume exist (along with their expected behavior):
+
+<CodeExample title="API Assumptions">
+
+```js
+// We assume that these to correspond to custodial account details.
+CUSTODIAL_KEY = StellarSdk.Keypair.fromSecret("...");
+custodialAcc = await server.loadAccount(CUSTODIAL_KEY.publicKey());
+
+/**
+ * Credits a customer (`to`) with `amount` of a particular `asset`.
+ *
+ * @param  {string}   to     A string representing the customer that's part of
+ *     this pooled account. This could be a key, a database index, an object,
+ *     or some other abstraction. The main point is that this is *NOT* a
+ *     Stellar account.
+ *
+ *     This value either directly comes from or somehow resolves from the
+ *     transaction memo field on the payment.
+ *
+ * @param  {string}   from   The public key of the Stellar account that
+ *                           submitted this deposit.
+ * @param  {Asset}    asset  The asset deposited into the customer's account.
+ * @param  {string}   amount The quantity (as a string, to conform to the way
+ *     payments are submitted on the Stellar network) of `asset` being
+ *     credited.
```

Comment:
Super NIT: the spacing in the param annotations for this function are slightly different than the annotations for the functions below. Totally not a big deal, but it just took me an extra second to parse

Reply:
I live for this level of scrutiny, ty!